src/main/resources/avro/reads.avdl

```diff
@@ -177,7 +177,38 @@ record LinearAlignment {
 }
 
 /**
-Each read alignment describes a linear alignment with additional information
+A string-to-reference-graph alignment can be represented by one CIGAR string and
+one `Path` through multiple `Reference`s, against which the CIGAR string is
+interpreted.
+
+Note that `Path`s are restricted to following reference adjacencies.
+If you want to align your read following adjacencies that are
+not present in the `ReferenceSet` to which you are aligning, you have a
+"chimeric" alignment, and should use multiple `ReadAlignment`s and the
+supplementaryAlignment flag instead of a single `GraphAlignment`.
+
+Some especially large deletions could be represented just as well as a large
+deletion in the CIGAR string, or as a chimeric alignment.
+*/
+record GraphAlignment {
```

Comment:
The current PR allows one to define a GraphAlignment which takes a Path through a non-reference (variant) segment. Is this desired behavior, or an unintended side-effect of using Segment for both References and Variants?

From my reading of the paper, and the comments in this PR, my understanding is that in a Graph-based model, Reads (for example) would still be aligned to a reference set, and not to a reference set + variant set(s).